packages/firestore/test/unit/specs/spec_test_runner.ts

```diff
@@ -1047,6 +1041,11 @@ abstract class TestRunner {
     }
   }
 
+  private validateSnapshotsInSyncEvents(count: number): void {
```

Comment:
I think this should be `number | undefined`?  And then you wouldn't need the `!` on line 525?
For clarity maybe rename `count` to `expectedCount`